x-pack/plugins/ml/public/application/jobs/jobs_list/components/utils.js

```diff
@@ -372,11 +372,11 @@ export function filterJobs(jobs, clauses) {
 // start datafeed modal.
 export function checkForAutoStartDatafeed() {
   const job = mlJobService.tempJobCloningObjects.job;
+  const datafeed = mlJobService.tempJobCloningObjects.datafeed;
   if (job !== undefined) {
     mlJobService.tempJobCloningObjects.job = undefined;
-    const hasDatafeed =
-      typeof job.datafeed_config === 'object' && Object.keys(job.datafeed_config).length > 0;
-    const datafeedId = hasDatafeed ? job.datafeed_config.datafeed_id : '';
+    const hasDatafeed = typeof datafeed === 'object' && Object.keys(datafeed).length > 0;
```

Comment:
`datafeed` should be set to `undefined` like `job` is on line 377.
This should also happen here:
https://github.com/elastic/kibana/blob/43faa76f676f9c8d808d2467209bc1de130f0237/x-pack/plugins/ml/public/application/jobs/new_job/pages/new_job/page.tsx#L90

Sorry, I missed this on the previous PR.

Reply:
Updated here https://github.com/elastic/kibana/pull/90078/commits/d867b4658ddd9fe855d3a5437ca242b8da329ae1